draco/analysis/transform.py

```diff
@@ -285,20 +295,20 @@ def pack_product_array(arr):
 
             # Generate weight
             if self.weight == 'inverse_variance':
-                wss = ss.weight[freq_ind, ss_pi]
+                wss = ssw[freq_ind, ss_pi]
 
             else:
-                wss = (ss.weight[freq_ind, ss_pi] > 0.0).astype(np.float32)
+                wss = (ssw[freq_ind, ss_pi] > 0.0).astype(np.float32)
                 wss *= nprod_in_stack[np.newaxis, ss_pi, stt:ett]
 
             # Accumulate visibilities, conjugating if required
             if feedconj == conj:
-                sp.vis[:, sp_pi] += wss * ss.vis[freq_ind, ss_pi]
+                spv[:, sp_pi] += wss * ssv[freq_ind, ss_pi]
             else:
-                sp.vis[:, sp_pi] += wss * ss.vis[freq_ind, ss_pi].conj()
+                spv[:, sp_pi] += wss * ssv[freq_ind, ss_pi].conj()
 
             # Accumulate variances in quadrature.  Save in the weight dataset.
-            sp.weight[:, sp_pi] += wss**2 * tools.invert_no_zero(ss.weight[freq_ind, ss_pi])
+            spw[:, sp_pi] += wss**2 * tools.invert_no_zero(ssw[freq_ind, ss_pi])
```

Comment:
maybe I'm just missing something but I don't see this accumulating variance. At least in the inverse variance case, doesn't this end up accumulating the weight (non-inverted)?

Reply:
It inverts the original weights to get the variances and then accumulates them. Maybe I'm not understanding your comment.

Reply:
oh I see, they are accumulated according to the weighted scheme... I was indeed missing something. It's just strange that in the inverse variance case you end up adding up the original weights